src/js/module/Clipboard.js

```diff
@@ -14,59 +39,50 @@ define([
      * @param {Event} event
      */
     var hPasteClipboardImage = function (event) {
+
       var clipboardData = event.originalEvent.clipboardData;
       var layoutInfo = dom.makeLayoutInfo(event.currentTarget || event.target);
       var $editable = layoutInfo.editable();
 
       if (!clipboardData || !clipboardData.items || !clipboardData.items.length) {
+
         var callbacks = $editable.data('callbacks');
         // only can run if it has onImageUpload method
         if (!callbacks.onImageUpload) {
           return;
         }
 
-        // save cursor
-        handler.invoke('editor.saveNode', $editable);
-        handler.invoke('editor.saveRange', $editable);
-
-        $editable.html('');
-
         setTimeout(function () {
-          var $img = $editable.find('img');
-
-          // if img is no in clipboard, insert text or dom
-          if (!$img.length || $img[0].src.indexOf('data:') === -1) {
-            var html = $editable.html();
-
-            handler.invoke('editor.restoreNode', $editable);
-            handler.invoke('editor.restoreRange', $editable);
-
-            handler.invoke('editor.focus', $editable);
-            try {
-              handler.invoke('editor.pasteHTML', $editable, html);
-            } catch (ex) {
-              handler.invoke('editor.insertText', $editable, html);
-            }
+          if (!pasteDiv) {
+            return;
+          }
+          
+          var $img = pasteDiv[0].childNodes[0];
+          if (!$img) {
             return;
           }
 
-          var datauri = $img[0].src;
+          handler.invoke('restoreRange', $editable);
+          if ($img.tagName !== 'IMG') {
```

Comment:
sometimes tagName is lowercase, It depends on browser, I thought `dom.isImg` is safe for this condition.
`if (!dom.isImg(imgNode))`
https://github.com/summernote/summernote/blob/develop/src/js/core/dom.js#L1037